src/FixedDataTableRow.js

```diff
@@ -175,40 +182,42 @@ class FixedDataTableRowImpl extends React.Component {
      */
     template: PropTypes.oneOf(['cell', 'footer', 'header']).isRequired,
   };
+  constructor(props) {
+    super(props);
+    this._initialRender = true;
+  }
+
+  componentDidMount() {
+    this._initialRender = false;
+  }
 
   shouldComponentUpdate(nextProps) {
     // only skip updates while scrolling
     if (!nextProps.isScrolling) {
       return true;
     }
 
-    // if row is not visible then no need to render it
-    // change in visibility is handled by the parent
+    // if row's visibility has changed, then update it
+    if (this.props.visible !== nextProps.visible) {
+      return true;
+    }
+
+    // if row is still not visible then no need to update
     if (!nextProps.visible) {
       return false;
     }
 
-    // Only update the row if scrolling leads to a change in horizontal offsets.
-    // The vertical offset is taken care of by the wrapper
+    // if these props haven't changed for the same row while scrolling, then skip update
     return !(
+      nextProps.isScrolling &&
       this.props.index === nextProps.index &&
+      this.props.offsetTop === nextProps.offsetTop &&
       this.props.scrollLeft === nextProps.scrollLeft
     );
   }
 
   render() /*object*/ {
     var subRowHeight = this.props.subRowHeight || 0;
-    var style = {
-      width: this.props.width,
-      height: this.props.height + subRowHeight,
-    };
-    var className = cx({
-      'fixedDataTableRowLayout/main': true,
-      'public/fixedDataTableRow/main': true,
-      'public/fixedDataTableRow/highlighted': this.props.index % 2 === 1,
-      'public/fixedDataTableRow/odd': this.props.index % 2 === 1,
-      'public/fixedDataTableRow/even': this.props.index % 2 === 0,
-    });
     var fixedColumnsWidth = this.props.fixedColumnsWidth;
```

Comment:
We will be adding all this styling and className in the function call (either of legacy or non-legacy)